app/src/main/java/de/grobox/transportr/favorites/trips/FavoriteTripPopupMenu.java

```diff
@@ -67,4 +80,21 @@ private void setFavState(MenuItem item, boolean is_fav) {
 		}
 	}
 
+	@Override
+	protected Intent getShortcutIntent(Context context, String type) {
+		// ignore type, insert uid
+		Intent shortcutIntent = new Intent(context, DirectionsActivity.class);
+		shortcutIntent.setAction(ACTION_SEARCH);
+		shortcutIntent.addFlags(FLAG_ACTIVITY_NEW_TASK);
+		shortcutIntent.addFlags(FLAG_ACTIVITY_CLEAR_TOP);
+		shortcutIntent.setData(Uri.parse(DirectionsActivity.INTENT_URI_FAVORITE));
+		shortcutIntent.putExtra(FAV_TRIP_UID, trip.getUid());
+		return shortcutIntent;
+	}
```

Comment:
Could you also pass in the proper type here and then do

```java
Intent shortcutIntent = super.getShortcutIntent(Context context, String type);
shortcutIntent.putExtra(FAV_TRIP_UID, trip.getUid());
return shortcutIntent;
```

This way, you can maybe get rid of the code duplication.

Reply:
Ok, now I am reusing some more of the methods which where actually in `SpecialLocationPopupMenu` in `AbstractFavoritesPopupMenu`, while most of them are still abstract and are overridden in the actual sub class. So all of them use `addShortcut()` and `getShortcutIntent()` with `FavoriteTripPopupMenu` adding the trips `uid`.